### DISCUSSION TITLE
Promote/Abort from argocd CLI or dashboard API

### DISCUSSION THREAD

**Discussion by mlewiarz** [argoproj/argo-rollouts#1686] [Q&A]
Hi, I would like to be able to promote/abort rollout through some API from my Gitlab CI jobs.
I found in docs that I can only use dashboard or kubectl plugin. 
For me both ways din't meet my needs so I used patches from plugin source code, it looks like:
`argocd app patch-resource $APP_NAME --kind Rollout --resource-name $ROLLOUT_NAME --patch '{"status":{"abort":true}}' --patch-type 'application/merge-patch+json'`
and
`argocd app patch-resource $APP_NAME --kind Rollout --resource-name $ROLLOUT_NAME--patch '{ "spec": { "paused": false }, "status": { "pauseConditions": null } }' --patch-type 'application/merge-patch+json'`

But it won't work from versions >0.9.0 so I'm stuck.
Maybe you can add to argocd CLI possibility to promote/abort or to dashboard some API?

**Comment by harikrongali** [comment]
@mlewiarz rollout has a plugin that you can use https://argoproj.github.io/argo-rollouts/features/kubectl-plugin/

**Comment by mlewiarz** [comment]
I wrote that I don't want to use kubectl plugin :) 
Perfect soultion for me would be integrating rollout promote/abort into argocd cli

**Comment by alexmt** [comment]
Hello @mlewiarz ,

It is already possible. Argo CD allows to interact with K8S resource using [Resource Actions](https://argo-cd.readthedocs.io/en/stable/operator-manual/resource_actions/). Rollouts actions are bundled by default. You can use the following command to promote rollout:

```
argocd app actions run <app-name> promote-full --kind Rollout
```

**Reply by alexmt** [reply]
and this command will print list of all available actions:

```
argocd app actions list <app-name> --kind Rollout
```
